### ISSUE TITLE
WTV file support?

### ISSUE THREAD

**Issue by Dan203** [MediaArea/MediaInfo#183]
I saw a conversation from a few years ago where you were looking at adding WTV file support, but it never seemed to come to fruition. The ffmpeg code has a pretty good WTV demuxer now. Is there any chance you could use that to start from to get it working in MediaInfo? 

I'd look at doing it myself but I've looked at the MediaInfo code and I don't even understand how it's structured. It would probably take me more time just to learn how to add something to the code then it would for you to just add it.

**Comment by JeromeMartinez** [comment]
Right, in the meanwhile FFmpeg implemented it so it would be easier for us to add the support by looking a bit at their code.
But there is still a lock of sponsor for this support, often requested but never funded (mainly because people requesting it are private persons not able to fund a feature alone), so it is still in the (long) list of features to do "for free" when we have free time.

Note that we plan to launch a crowdfunding campaign soon, and participants will be able to prioritize requests so we'll see if WTV support is the priority in features we suggest to implement with the campaign.

**Comment by JeromeMartinez** [comment]
Unfortunately there is current a bit lack of documentation.
But expanding the [WTV parser file](https://github.com/MediaArea/MediaInfoLib/blob/master/Source/MediaInfo/Multiple/File_Wtv.cpp) (nearly empty for the moment) with the file header would be a first step.

Even without binding frames to another parser (e.g. video, audio...), you can see the result of your changes with the [MediaTrace](https://mediaarea.net/MediaTrace) feature (with command line: `--Details=1` and with Windows GUI menu Debug, Details set to 10 then parse the file), then it would be easier for me to fill MediaInfo fields.
For expanding more (i.e. not only the file header), [this example](https://github.com/MediaArea/MediaInfoLib/blob/master/Source/MediaInfo/Multiple/File_Rm.cpp) may help understanding how to use Header_Parse and Data_Parse virtual methods (called in a loop) and implementing the workflow (frame header, frame content) would make easy for me to bind to the corresponding parser (MPEG-TS parser? or MPEG video parser?)

**Comment by JeromeMartinez** [comment]
You can vote for this feature: 
[Vote for MediaInfo WTV support](https://mediaarea.net/Vote/Wtv-Support-51).
